Tests/TocTests.fs

```diff
@@ -75,13 +152,79 @@ module RenderToc =
         let expectedLines =
             [| Toc.StartMarker
                "- [T1](#t1)"
-               " - [T2](#t2)"
-               "  - [T3](#t3)"
-               " - [T4](#t4)"
-               "  - [T5](#t5)"
+               "  - [T2](#t2)"
+               "    - [T3](#t3)"
+               "  - [T4](#t4)"
+               "    - [T5](#t5)"
                Toc.EndMarker
                "" |]
 
         let expected = String.concat NewLine expectedLines
 
         Assert.Equal(expected, titles)
+
+module DocumentEdit =
+    [<Fact>]
+    let insert_afterYaml () =
+        let doc =
+            makeFakeDocumentLines
+                [| "---"
```

Comment:
Do you wanna try raw strings maybe https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/strings#verbatim-strings ? You can also interpolate them if needed https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/interpolated-strings#verbatim-interpolated-strings